tests/conftest.py

```diff
@@ -117,6 +117,21 @@ def chdir(path):
 #
 
 
+@pytest.fixture(autouse=True)
+def setup_teardown_test_environment(tmpdir):
```

Comment:
Come to think of it, we can even merge this fixture with `independent_global_and_system_config`.